src/Client.ts

```diff
@@ -177,54 +203,160 @@ export class Client extends ClientJS {
           this.decorators.trigger(on.event, this, true)
         );
       } else {
-        this.on(
-          on.event as any,
-          this.decorators.trigger(on.event, this)
-        );
+        this.on(on.event as any, this.decorators.trigger(on.event, this));
       }
     });
 
-    return await super.login(token);
+    return super.login(token);
   }
 
   /**
    * Initialize all the @Slash with their permissions
    */
-  async initSlashes() {
-    await Promise.all(
-      this.slashes.map(async (slash) => {
-        // Init all the @Slash
-        if (slash.guilds.length > 0) {
-          // If the @Slash is guild specific, add it to the guild
-          await Promise.all(
-            slash.guilds.map(async (guildID) => {
-              const guild = this.guilds.cache.get(guildID as Snowflake);
-
-              if (!guild) {
-                throw new GuildNotFoundError(guildID);
-              }
-
-              const commands = guild.commands;
-              const command = await commands.create(slash.toObject());
-
-              if (slash.permissions.length <= 0) return;
-
-              await commands.setPermissions(command, slash.getPermissions());
-            })
-          );
-        } else {
-          // If the @Slash is global, add it globaly
-          const commands = this.application.commands;
-          const command = await commands.create(slash.toObject());
+  async initSlashes(options?: {
+    log: { forGuild: boolean; forGlobal: boolean };
+  }) {
+    // # first init guild's all slash
+    const guildSlashStorage = new Map<string, DSlash[]>();
```

Comment:
Can we give these 2 vars better names? Are these both for guild specific slashes, or are these also used for global slashes?

Reply:
that is only for guild slashes, global does not need it.